src/status_im2/contexts/chat/messages/drawers/view.cljs

```diff
@@ -0,0 +1,144 @@
+     :label    (i18n/label :t/copy-text)
+     :icon     :i/copy
+     :id       :copy}]
+   (when message-pin-enabled
+     [{:type     :main
+       :on-press #(pin-message message-data)
+       :label    (i18n/label (if pinned
+                               (if community? :t/unpin-from-channel :t/unpin-from-chat)
+                               (if community? :t/pin-to-channel :t/pin-to-chat)))
+       :icon     :i/pin
+       :id       (if pinned :unpin :pin)}])
+   (when-not pinned
+     [{:type     :danger
+       :on-press #(rf/dispatch [:chat.ui/delete-message-for-me message-data constants/delete-message-for-me-undo-time-limit-ms])
+       :label    (i18n/label :t/delete-for-me)
+       :icon     :i/delete
+       :id       :delete-for-me}])
+   (when (and (or outgoing can-delete-message-for-everyone?) config/delete-message-enabled?)
+     [{:type     :danger
+       :on-press #(rf/dispatch [:chat.ui/delete-message message-data constants/delete-message-undo-time-limit-ms])
+       :label    (i18n/label :t/delete-for-everyone)
+       :icon     :i/delete
+       :id       :delete-for-all}])))
+
+(defn reactions [message-id]
+  [rn/view {:style {:flex-direction     :row
+                    :justify-content    :space-between
+                    :padding-horizontal 30
+                    :padding-top        5
+                    :padding-bottom     15}}
+   (doall
+    (for [[id icon] constants/reactions]
+      ;:let [active (own-reactions id)]]
+      ;;TODO reactions selector should be used https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=9961%3A166549
+      ;; not implemented yet
+      ^{:key id}
+      [quo/button (merge
+                   {:size                40
+                    :type                :grey
+                    :icon                true
+                    :icon-no-color       true
+                    :accessibility-label (str "emoji-picker-" id)
+                    :on-press            #(do
+                                            (rf/dispatch [:models.reactions/send-emoji-reaction
+                                                          {:message-id message-id
+                                                           :emoji-id   id}])
+                                            (rf/dispatch [:bottom-sheet/hide]))})
+       ;(when active {:style {:background-color colors/neutral-10}}))
+       icon]))])
+
+;; TODO (flexsurfer) this probably should be a component in quo2 https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System?node-id=5626%3A158317&t=1JiRhBswpVXZ4SYF-3
+(defn reactions-and-actions [message-data]
+  (fn []
+    (let [actions (get-actions message-data)
+          main-actions (filter #(= (:type %) :main) actions)
+          danger-actions (filter #(= (:type %) :danger) actions)
+          admin-actions (filter #(= (:type %) :admin) actions)]
+      [:<>
+       ;; REACTIONS
+       [reactions (:message-id message-data)]
+
+       ;; MAIN ACTIONS
```

Comment:
I think better than these uppercase comments saying WHAT is being done, you could extract the *main actions* piece to a separate function, the *danger actions* to a separate function and so on.

The idea being that self-documenting executable code is better than comments saying what they do. Proper function names are self-documenting.